feat: adds other framework docs (#663)
supertokens/docs: v2/thirdpartyemailpassword/other-frameworks.mdx

---
id: other-frameworks
title: "Supporting other backend frameworks"
hide_title: false
---

If your backend framework is not supported by SuperTokens, you can follow this guide on setting up an authentication server to protect your frontend and backend.

A key feature of the SuperTokens backend SDK is the middleware it adds to your backend server. This middleware will add authentication routes to handle requests like user creation and login. If your backend framework is not supported by SuperTokens, you will need to run a separate server with the SuperTokens backend SDK configured. This could be a service in NodeJS, Python or Golang.

> To illustrate how this setup works, lets take an example:

Reviewer: ```suggestion
To illustrate how this setup works, let's take an example:
```

> - The user has a React based frontend and a php backend server.

Reviewer: ```suggestion
- The user has a React based frontend and a PHP backend server.
```

- SuperTokens does not have a php SDK, so we spin up a separate service in NodeJS to act as an authentication server. This server will be configured with the SuperTokens backend SDK to handle authentication requests and the issuing of access and refresh tokens.
- You can follow our [quick setup guide on setting up the NodeJS server](./pre-built-ui/setup/backend).
- The architecture of setup should have the following design:

![SuperTokens Setup With Authentication server](/img/other-framework-setup.png)

- In the architecture above the the React Frontend communicates with the php and node server through a reverse proxy.

> - Authentication requests like sign-up or sign-in are routed to the NodeJS server while regular requests are sent to the php server.

Reviewer: ```suggestion
- Authentication requests like sign-up or sign-in are routed to the NodeJS server. These APIs are automatically created and handled by our supertokens-node SDK. Your application specific APIs are sent to the PHP server.
```

> - Incoming requests to the php server can be validated by [verifying the JWT sent along with the request](./common-customizations/sessions/with-jwt/jwt-verification).

Reviewer: Clearly mention about session creation

:::important

> The method mentioned above assumes that the authentication server is running on the same domain as the php server. If your authentication server runs on a separate subdomain, you will need to [enable cookie sharing](./common-customizations/sessions/multiple-api-endpoints) so that cookies are automatically attached to requests to the php server.

Reviewer: ```suggestion
The method mentioned above assumes that the authentication server is running on the same domain as the php server. If instead, your authentication server runs on a separate subdomain, you will need to [enable cookie sharing](./common-customizations/sessions/multiple-api-endpoints) so that cookies are automatically attached to requests to the php server.
```

> :::

Reviewer: add an explicit point aobut session refreshin. That when the access token is missing or has expired, the php server should send back a 401, which will ...
